docs/advanced/node-local-storage.md

```diff
@@ -0,0 +1,356 @@
+
+> When first testing local volumes, we wanted to have a thorough understanding of
+> the effect disruptions (voluntary and involuntary) would have on pods using
+> local storage, and so we began testing some failure scenarios. We found that
+> when a local volume becomes unavailable while the node remains available (such
+> as when performing maintenance on the disk), a pod using the local volume will
+> be stuck in a ContainerCreating state until it can mount the volume. If a node
+> becomes unavailable, for example if it is removed from the cluster or is
+> drained, then pods using local volumes on that node are stuck in an Unknown or
+> Pending state depending on whether or not the node was removed gracefully.
+
+> Recovering pods from these interim states means having to delete the PVC binding
+> the pod to its local volume and then delete the pod in order for it to be
+> rescheduled (or wait until the node and disk are available again). We took this
+> into account when building our operator for M3DB, which makes changes to the
+> cluster topology when a pod is rescheduled such that the new one gracefully
+> streams data from the remaining two peers. Eventually we plan to automate the
+> deletion and rescheduling process entirely.
+>
+> Alerts on pod states can help call attention to stuck local volumes, and
+> workload-specific controllers or operators can remediate them automatically.
+> Because of these constraints, it’s best to exclude nodes with local volumes from
+> automatic upgrades or repairs, and in fact some cloud providers explicitly
+> mention this as a best practice.
+
+And some information of the current pain points that might be handled better in
+the future:
+
+> One of the most frequent asks has been for a controller that can help with
+> recovery from failed nodes or disks, which is currently a manual process (or
+> something that has to be built into an operator). SIG Storage is investigating
+> creating a common controller that can be used by workloads with simple and
+> similar recovery processes.
+>
+> Another popular ask has been to support dynamic provisioning using lvm. This can
+> simplify disk management, and improve disk utilization. SIG Storage is
+> evaluating the performance tradeoffs for the viability of this feature.
+
+
+## Local Persistent Volumes in Raw Block Device Mode
+
+This mode of using node local storage has not been explored so far, but it's
+listed so it is explored in the future.
+
+Some links that might be relevant when doing so are left:
+
+https://kubernetes.io/blog/2019/03/07/raw-block-volume-support-to-beta/
+https://kubernetes.io/docs/concepts/storage/volumes/#local
+https://docs.okd.io/latest/install_config/persistent_storage/persistent_storage_local.html
+https://docs.okd.io/latest/install_config/configuring_local.html#install-config-configuring-local
+https://docs.okd.io/latest/install_config/configuring_local.html#local-volume-raw-block-devices
+
+
+## Using OpenEBS Local PV
```

Comment:
Like I said earlier in the previous comment this deserves its own document.

Reply:
I prefer to start with a single document of node local storage, if at a later stage we want to have what are currently sections in different documents we can do when we add more stuff. What do you think?

Reply:
I want to have a simple and consice documentation for the users so that we can just point them to do certain task. Like for manual node local PV I would point someone to this doc. The doc should not confuse user.

Reply:
Can you please elaborate on why would this confuse the user? I only see an almost empty section that says "we will expand on this later, these links might be relevant for that".

Would an empty section or removing this be better from your point of view? I'm not sure what to do to address your comment :-/. An new doc with this links doesn't seem great (it's pretty much empty) and not sure what other ways are to address this.

Reply:
@surajssd can you look into the modified version? I think it's fine for now to keep it here until we have something more specific. Then we can just refer here to new document.